modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/http4s/SeqexecUIApiRoutes.scala

```diff
@@ -116,15 +119,24 @@ class SeqexecUIApiRoutes(devMode: Boolean, auth: AuthenticationService, events:
           case NullEvent => false
           case _         => true
         }
+        def filterOutOnClientId(clientId: ClientID) = (e: SeqexecEvent) => e match {
+          case e: ForClient if e.clientId =/= clientId => false
+          case _                                       => true
+        }
         // If the user didn't login, anonymize
         val anonymizeF: SeqexecEvent => SeqexecEvent = user.fold(_ => anonymize _, _ => identity _)
-        WS(
-          Exchange(
-            Process.emit(Binary(trimmedArray(ConnectionOpenEvent(user.toOption)))) ++
-              (pingProcess merge engineOutput.subscribe.map(anonymizeF).filter(filterOutNull).map(v => Binary(trimmedArray(v)))),
-            scalaz.stream.Process.empty
-          )
-        )
+        // Create a client specific process
+        for {
+          clientId <- Task.delay(UUID.randomUUID())
```

Comment:
I'm curious why you wrapped the `UUID.randomUUID()` call in a `Task.delay`.

(I'm sure there's a good reason for it. I'm just not sure why you want to delay the creation of the random UUID.)

Reply:
`Task` wraps the call to `randomUUID` making it pure and allowing me to return a pure `Task` for http4s to reuse

Reply:
👍 
I figured it was something like that, but I wasn't sure.